kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java

```diff
@@ -446,6 +447,162 @@ public void buildIpv6SettingsShouldThrowIllegalArgumentExceptionWhenGivenExpecte
         thenIllegalArgumentExceptionThrown();
     }
 
+    @Test
+    public void build8021xSettingsShouldThrowIfIsEmpty() {
+        givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+
+        whenBuild8021xSettingsIsRunWith(this.networkProperties, "wlan0");
+
+        thenNoSuchElementExceptionThrown();
+    }
+
+    @Test
+    public void build8021xSettingsShouldThrowIfEapIsInvalid() {
+        givenMapWith("net.interface.wlan0.config.802-1x.eap", "invalid eap value");
+        givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+
+        whenBuild8021xSettingsIsRunWith(this.networkProperties, "wlan0");
+
+        thenIllegalArgumentExceptionThrown();
+    }
+
+    @Test
+    public void build8021xSettingsShouldThrowIfInnerAuthIsInvalid() {
+        givenMapWith("net.interface.wlan0.config.802-1x.eap", "invalid eap value");
+        givenMapWith("net.interface.wlan0.config.802-1x.innerAuth", "invalid Inner Auth value");
+        givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+
+        whenBuild8021xSettingsIsRunWith(this.networkProperties, "wlan0");
+
+        thenIllegalArgumentExceptionThrown();
+    }
+
+    @Test
+    public void build8021xSettingsTtlsAndMschapV2() {
+        givenMapWith("net.interface.wlan0.config.802-1x.eap", "ttls");
+        givenMapWith("net.interface.wlan0.config.802-1x.innerAuth", "mschapv2");
+        givenMapWith("net.interface.wlan0.config.802-1x.identity", "example-user-name");
+        givenMapWith("net.interface.wlan0.config.802-1x.password", new Password("secure-test-password-123!@#"));
+        givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+
+        whenBuild8021xSettingsIsRunWith(this.networkProperties, "wlan0");
+
+        thenNoExceptionsHaveBeenThrown();
+
+        thenResultingMapContainsArray("eap", new Variant<>(new String[] { "ttls" }).getValue());
+        thenResultingMapContains("phase2-auth", "mschapv2");
+        thenResultingMapContains("identity", "example-user-name");
+        thenResultingMapContains("password", "secure-test-password-123!@#");
+        thenResultingMapNotContains("anonymous-identity");
+        thenResultingMapNotContains("ca-cert");
+        thenResultingMapNotContains("ca-cert-password");
+
+    }
+
+    @Test
+    public void build8021xSettingsTtlsAndMschapV2WithOptionalParams() {
```

Comment:
Please use the established test naming convention.

```suggestion
    public void build8021xSettingsShouldWorkWithTtlsAndMschapV2AndOptionalParams() {
```